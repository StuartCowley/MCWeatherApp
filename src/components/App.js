import React, { useState, useEffect } from "react";
import getForecasts from "../requests/getForecasts";

import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import DetailedSummary from "./DetailedSummary";
import SearchForm from "./SearchForm";

import "../styles/App.css";

const App = () => {
  const [forecasts, setForecasts] = useState([]);
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [searchText, setSearchText] = useState("");

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const handleLocationChange = () => {
    getForecasts(setForecasts, setLocation, setSelectedDate, searchText);
  };

  useEffect(() => {
    getForecasts(setForecasts, setLocation, setSelectedDate);
  }, []);

  const { city, country } = location;

  return (
    <div className="weather-app">
      <LocationDetails city={city} country={country} />
      <SearchForm
        onChangeLocation={handleLocationChange}
        updateSearchText={setSearchText}
      />
      <ForecastSummaries
        forecasts={forecasts}
        onForecastSelect={handleForecastSelect}
      />
      {selectedForecast && <DetailedSummary forecast={selectedForecast} />}
    </div>
  );
};

export default App;
