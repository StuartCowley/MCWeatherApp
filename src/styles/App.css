.weather-app {
    margin-left: 1rem;
    margin-right: 1rem;
}