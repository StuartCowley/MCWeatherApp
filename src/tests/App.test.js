import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";
import { location, forecasts } from "../data/placeholderData.json";

test("renders weather app heading", () => {
  render(<App location={location} forecasts={forecasts} />);
  const linkElement = screen.getByText(/forecast summaries/i);
  expect(linkElement).toBeInTheDocument();
});
