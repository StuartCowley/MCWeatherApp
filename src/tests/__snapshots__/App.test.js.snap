// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1
      class="location-details"
    >
      Forecast for , 
    </h1>
    <div>
      <form>
        <input
          placeholder="Enter city"
          type="text"
        />
        <button
          class="search-form__button"
          type="submit"
        >
          Change location
        </button>
      </form>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
